test-fixtures/src/test/java/org/sonar/api/testfixtures/log/LogTesterTest.java

```diff
@@ -51,8 +51,16 @@ public void info_level_by_default() {
   }
 
   @Test
-  public void dont_set_info_level_if_it_was_changed() {
+  public void remember_default_level_if_it_was_changed_before_first_test() {
+    // Change default level to DEBUG instead of INFO by default
     LogTester underTest = new LogTester().setLevel(Level.DEBUG);
+    // First test
+    underTest.before();
```

Comment:
isn't there a wrong assumption that the test class is not re-instantiated between tests?

Reply:
This is indeed the default behavior, but this can be changed (at least in JUnit 5) with `Lifecycle.PER_CLASS` (see https://junit.org/junit5/docs/current/user-guide/#writing-tests-test-instance-lifecycle)